### DISCUSSION TITLE
Don't work url launch in android

### DISCUSSION THREAD

**Discussion by SmartDevStar** [community/community#57148] [Mobile] [answered]
### Select Topic Area

Question

### Body

I am going to launch url in android app using flutter.
This is the code I've written to implement url launch.
onPressed: () async {
              String msteamsUrl =
                  _activeClass.sessionWebLink!.replaceFirst("https", "msteams");
              final uri = Uri.parse(msteamsUrl);
              if (await canLaunchUrl(uri)) {
                await launchUrl(uri);
              } else {
                ScaffoldMessenger.of(context).hideCurrentSnackBar();
                ScaffoldMessenger.of(context).showSnackBar(const SnackBar(
                  content: Text(
                    "First install Microsoft Teams app..",
                    style: TextStyle(
                      color: Colors.red,
                    ),
                  ),
                ));
              }
            },
         But when I tab the launch button, "Webpage not available" is displayed. How can I solve this problem?

**Comment by SuperDev314** [comment, marked as answer]
Hi, Christopher, I've faced the same issues before and I've solved the problem using this code.

String msteamsUrl =  _activeClass.sessionWebLink!.replaceFirst("https", "msteams");
final uri = Uri.parse(msteamsUrl);
              if (await canLaunchUrl(uri)) {
                await launchUrl(uri);
              } else {
                ScaffoldMessenger.of(context).hideCurrentSnackBar();
                ScaffoldMessenger.of(context).showSnackBar(const SnackBar(
                  content: Text(
                    "First install Microsoft Teams app..",
                    style: TextStyle(
                      color: Colors.red,
                    ),
                  ),
                ));
              }
and you should install  url_launcher: ^6.1.11 package in you project.

**Reply by SmartDevStar** [reply]
Thanks, It does work on my project.
That's very exactly!!
